services/issue/pull.go

```diff
@@ -90,7 +90,7 @@ func PullRequestCodeOwnersReview(ctx context.Context, pull *issues_model.Issue,
 
 	// https://github.com/go-gitea/gitea/issues/29763, we need to get the files changed
 	// between the merge base and the head commit but not the base branch and the head commit
-	changedFiles, err := repo.GetFilesChangedBetween(mergeBase, pr.HeadCommitID)
+	changedFiles, err := repo.GetFilesChangedBetween(mergeBase, pr.GetGitRefName())
```

Comment:
I am not sure whether it is right for a forked repo.

If there are too many unclear things, I would suggest to ignore some non-important errors (only panic in dev mode), don't block 1.21 users.

Reply:
See my comment https://github.com/go-gitea/gitea/issues/29997#issuecomment-2015040562

Anything wrong in `PullRequestCodeOwnersReview` shouldn't block user from changing the title.

Reply:
For a forked repository, it's still right. The file `refs/pull/<id>/head` stored the pull request's head commit id. It's right even the branch has been removed.

Reply:
But what if it doesn't exist? I do not any error in PullRequestCodeOwnersReview should block user's other operations like changing the title.

I prefer the error to be ignored but not reporting a 500 error page.

Reply:
02833022b2569e9970b2f6dff22682f8308ea94f